thunder/core/transforms.py

```diff
@@ -1042,9 +1042,24 @@ def _div_prim_grad(a: Number | TensorProxy, b: Number | TensorProxy, /) -> Numbe
 
 register_grad(pids.DIV, _div_prim_grad)
 
+
+# NOTE not differentiable, but it would trigger a flatten failure without a grad function
+# NOTE that's probably a bad error message that we should fix.
+@torchctx
+def _py_floordiv_prim_grad(a: Number | TensorProxy, b: Number | TensorProxy, /) -> Number | TensorProxy:
```

Comment:
want to point out a somewhat orthogonal changes here.

Is it expected that changing things to NumberProxy would push more requirement on having a grad rule?

Reply:
I think @IvanYashchuk has had some thoughts about producing gradients for numbers, and not just tensors.

fyi @IvanYashchuk to review this particular change

@jjsjann123, is the correct interpretation of this grad rule that it just stops grad from going from the output of the floor division to its inputs?

Are we calling floor division on tensors now, btw? Is the comment here:

https://github.com/Lightning-AI/lightning-thunder/blob/eb6565b15ed8f199c4cc8e1b2fa91d256aa8d975/thunder/core/prims.py#L2311

out of date?

Reply:
> @jjsjann123, is the correct interpretation of this grad rule that it just stops grad from going from the output of the floor division to its inputs?

yes. Since we are not supposed to propagate it back and are not expecting tensor inputs as well.

> Are we calling floor division on tensors now, btw? 

No we are *NOT* calling floordiv with tensors. I can update the annotation if you think we should restrict it. And I should probably also put an assert there that a and b are not TensorProxy.

The reason we needed this is for
`test_phantom_grad_vs_torch_consistency_reshape_`

sequences are not baked in, so our reshape handling of `-1` shows up in trace.

e.g. something like this vvv.
```
@torch.no_grad()
@no_autocast
def reshape(a, shape):
  # a: "cpu f16[4, 2]"
  # shape: "Collection"
  i0, _, i2, = shape
  i5 = prims.mul(1, i0)  # i5: "int 2"
  i9 = prims.mul(i5, i2)  # i9: "int 4"
  i12 = prims.py_floordiv(8, i9)  # i12: "int 2"
  t20 = prims.reshape(a, (i0, i12, i2))  # t20: "cpu f16[2, 2, 2]"
  return t20
```

Reply:
looks like the tests are still using `thunder.compile`, which handles sequence differently than `thunder.jit` that bakes in scalar values in a sequence. I think this explains why I have to navigate through the CI for that.

Reply:
A year ago Thunder supported differentiating function with operations on numbers and then it got broken. But it's only for floats. Non-differentiable inputs should be taken care as constants, so it's a quick of current transform code.
Could you please try this again to check if this "rule" is still needed after https://github.com/Lightning-AI/lightning-thunder/pull/364 is merged?

Reply:
I removed this piece of code and the tests still passed.
```
pytest thunder/tests/test_grad.py -k "test_phantom_grad_vs_torch_consistency_reshape_"
```

Reply:
Ha thanks a lot. I'll clean it up.